src/dbnode/persist/fs/files.go

```diff
@@ -1231,7 +1199,62 @@ func SortedCommitLogFiles(commitLogsDir string) ([]string, error) {
 	return sortedCommitLogFiles(commitLogsDir, commitLogFilePattern)
 }
 
+type filesetFile struct {
+	volumeIndex int
+	blockStart  xtime.UnixNano
+	fileName    string
+}
+
 type toSortableFn func(files []string) sort.Interface
+type toBlockStartAndVolumeIndexFn func(file string) (xtime.UnixNano, int, error)
+type sortedFilesetFiles []filesetFile
+
+func (s sortedFilesetFiles) Len() int {
+	return len(s)
+}
+
+func (s sortedFilesetFiles) Less(i, j int) bool {
+	ti := s[i].blockStart
+	tj := s[j].blockStart
+
+	if ti.Before(tj) {
+		return true
+	}
+
+	ij := s[j].volumeIndex
+	ii := s[i].volumeIndex
+	return ti.Equal(tj) && ii < ij
+}
+
+func (s sortedFilesetFiles) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+
+func findFilesetFiles(fileDir string, pattern string, fn toBlockStartAndVolumeIndexFn) (sortedFilesetFiles, error) {
+	matched, err := filepath.Glob(path.Join(fileDir, pattern))
+	if err != nil {
+		return nil, err
+	}
+	if len(matched) == 0 {
+		return nil, nil
+	}
```

Comment:
I think this `if` is redundant.

Reply:
I didn't want to change semantics on what is returned from this function. Previously `nil` would be returned if  `filepath.Glob` returns `nil`. Without this `if` check we would be returning empty slice which is not the same thing as before.

Reply:
`findSortedFilesetFiles` is a new function so I guess there can be no existing semantics for it.
And for the `filesetFiles`, there is still an `if` for this purpose:
https://github.com/m3db/m3/pull/3900/files#diff-78d9cf687193bca4cdc4ae73e54059f6a3b3ab4360a627c4bf26c070b8a0f909R1340

Reply:
`findSortedFilesetFiles` replaced `findFiles` (which is still present) so its semantics were based on it so that these both functions could be used interchangeably if needed. In Go returning nil slices is quite common and I don't see problems with this approach here.

Reply:
But `findFiles` does not have such check, it returns whatever is returned by `filepath.Glob`...

Reply:
Exactly, and `filepath.Glob` returns `nil` if it founds nothing. So `findSortedFilesetFiles`should also return `nil` when it founds nothing.